configure.ac: add check for NO_GETCONTEXT definition

Signed-off-by: Samuel Martin <s.martin49@gmail.com>
[yann.morin.1998@free.fr: add a comment, change variable name, use
 AS_IF, remove debug traces, use AC_CHECK_FUNCS (as suggested by
 Thomas)]
Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>
Cc: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

---
Upstream-Status: Pending
 configure.ac | 6 ++++++
 1 file changed, 6 insertions(+)

--- bdwgc-7.2f.orig/configure.ac	2014-06-01 19:00:47.000000000 +0200
+++ bdwgc-7.2f/configure.ac	2014-12-23 14:13:11.585716713 +0100
@@ -365,6 +365,12 @@
   AC_MSG_RESULT($ac_cv_fno_strict_aliasing)
 fi
 
+# Check for getcontext (uClibc can be configured without it, for example)
+AC_CHECK_FUNCS([getcontext])
+AS_IF([test "$ac_cv_func_getcontext" = "no"],
+  [CFLAGS="$CFLAGS -DNO_GETCONTEXT"
+   CPPFLAGS="$CPPFLAGS -DNO_GETCONTEXT"])
+
 case "$host" in
 # While IRIX 6 has libdl for the O32 and N32 ABIs, it's missing for N64
 # and unnecessary everywhere.
